### PR TITLE
docs: place demos at the top of each docs page

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
Update docs content to show a demo of the component under documentation at the beginning of the page.
- update to 11ty @ 3.0 (a pre-release version) so that ESM can be used across all of the data files in the build process
- structure the demo to build from `*/stories/args.ts` and `*/stories/template.ts` files
- hide NPM, minzipped and "Try it on" links from each README.md
- add webcomponents.dev and Spectrum spec URLs to documentation build data
- power the listings of the above data with scripts to scrape it from README.md files or metadata.yml files
- prepare various variants of the component page
  - component + demo + variables (features all three new sections)
  - component + demo (does not include the "controls" section)
  - component only (features only the Cards, which are dynamic to the presence of appropriate links)

**Follow up work**
- This removes the "minzipped" size data, which was always close enough but not 100% correct. We likely should replace it some how.
- We could checksize at build time be leveraging the benchmark tests as "contained" applications for measurement.
- The benchmark tests all rely on the theme system which means they are larger that _just_ the component, but as large as you'd likely need to use the component.
- all of the packages need `template.ts` and `args.ts` files to support this style of documentation.

## Related issue(s)
- refs #1822
  - I proposed we either...
    - land this, and then do individual component PRs. OR
    - count this as a project branch and submit individual components to it before merging.

## How has this been tested?

-   [ ] _Test case 1_
    1. Go [here](https://demo-first-docs--spectrum-web-components.netlify.app/components/action-bar/)
    2. See a docs page with component + demo + variables
-   [ ] _Test case 2_
    1. Go [here](https://demo-first-docs--spectrum-web-components.netlify.app/components/accordion/)
    2. See a docs page with component + demo + variables, but not all cards
-   [ ] _Test case 3_
    1. Go [here](https://demo-first-docs--spectrum-web-components.netlify.app/components/button/)
    2. See a docs page with component + demo
-   [ ] _Test case 4_
    1. Go [here](https://demo-first-docs--spectrum-web-components.netlify.app/components/avatar/)
    2. See a docs page with component and all cards
-   [ ] _Test case 5_
    1. Go [here](https://demo-first-docs--spectrum-web-components.netlify.app/components/badge/)
    2. See a docs page with component and some cards

## Types of changes
-   [x] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist
-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Rajdeepc** [comment]
I loved the preview mode and the approach of UI. Great addition to the docs page. Thank you!

**Comment by najikahalsema** [comment]
Would it be possible to get a link/card to the storybook, as well? I think that would be great for our consumers.

**Comment by Westbrook** [comment]
> Would it be possible to get a link/card to the storybook, as well? I think that would be great for our consumers.
Make a lot of sense! I added an issue, so it could be follow up work on this addition. See #4008

File: projects/documentation/content/_includes/partials/demo.njk [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            </sp-picker>
+        </div>
+    {% endfor %}</sp-field-group>{% endif %}
+    <div
+        class="usage-links"
+    >
+        <sp-card
+            size="s"
+            horizontal
+            href="https://www.npmjs.com/package/@spectrum-web-components/button"
```
Comment (Westbrook) [inline projects/documentation/content/_includes/partials/demo.njk:33]: Make this dynamic.

**Review by Rajdeepc — APPROVED** [review]
Awesome!

**Comment by TarunAdobe** [comment]
Just an observation, Currently the args aren't setup for any of those updates stories... are we planning to do that separately in other PRs or should we do that together with this? @Westbrook

**Pull request merged**